dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-fee-record-correction-transient-form-data.controller/index.test.ts

```diff
@@ -0,0 +1,187 @@
+import httpMocks, { MockResponse } from 'node-mocks-http';
+import { ObjectId } from 'mongodb';
+import {
+  FeeRecordCorrectionTransientFormDataEntityMockBuilder,
+  RECORD_CORRECTION_REASON,
+  RecordCorrectionTransientFormData,
+  TestApiError,
+} from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { Response } from 'express';
+import { GetFeeRecordCorrectionTransientFormDataRequest, GetFeeRecordCorrectionTransientFormDataResponse, getFeeRecordCorrectionTransientFormData } from '.';
+import { FeeRecordCorrectionTransientFormDataRepo } from '../../../../../repositories/fee-record-correction-transient-form-data-repo';
+import { FeeRecordRepo } from '../../../../../repositories/fee-record-repo';
+
+jest.mock('../../../../../repositories/fee-record-repo');
+jest.mock('../../../../../repositories/fee-record-correction-transient-form-data-repo');
+
+console.error = jest.fn();
+
+describe('get-fee-record-correction-transient-form-data.controller', () => {
+  describe('getFeeRecordCorrectionTransientFormData', () => {
+    const reportId = 1;
+    const feeRecordId = 2;
+
+    const tfmUserId = new ObjectId().toString();
+
+    const aValidRequestQuery = () => ({ reportId: reportId.toString(), feeRecordId: feeRecordId.toString(), userId: tfmUserId });
+
+    const mockFeeRecordExists = jest.fn();
+    const mockTransientFormDataFind = jest.fn();
+
+    let req: GetFeeRecordCorrectionTransientFormDataRequest;
+    let res: MockResponse<Response>;
+
+    beforeEach(() => {
+      FeeRecordRepo.existsByIdAndReportId = mockFeeRecordExists;
+      FeeRecordCorrectionTransientFormDataRepo.findByUserIdAndFeeRecordId = mockTransientFormDataFind;
+
+      req = httpMocks.createRequest<GetFeeRecordCorrectionTransientFormDataRequest>({
+        params: aValidRequestQuery(),
+      });
+      res = httpMocks.createResponse();
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it(`should respond with a '${HttpStatusCode.Ok}' and the retrieved form data if a transient form data entity exists`, async () => {
+      // Arrange
+      const formData: RecordCorrectionTransientFormData = {
+        reasons: [RECORD_CORRECTION_REASON.OTHER],
+        additionalInfo: 'Some additional information',
+      };
+      const transientFormDataEntity = new FeeRecordCorrectionTransientFormDataEntityMockBuilder().withFormData(formData).build();
+
+      mockFeeRecordExists.mockReturnValue(true);
```

Comment:
this could probably go in the beforeEach and then you only need it in the tests where it is not true

Reply:
This is related to one of Ruby's comments for another mock - https://github.com/UK-Export-Finance/dtfs2/pull/4022/files/ec1f46f38511503b471924a97ede4396e76c4c0b#r1867570291

I think as only 4 of the 11 tests rely on `mockFeeRecordExists` returning true, it's good to have it only in those individual test cases